store/src/main/java/org/apache/rocketmq/store/queue/RocksDBConsumeQueueTable.java

```diff
@@ -0,0 +1,312 @@
+        boolean min) throws RocksDBException {
+        long resultCQOffset = -1L;
+        long resultPhyOffset = -1L;
+        while (high >= low) {
+            long midCQOffset = low + ((high - low) >>> 1);
+            ByteBuffer byteBuffer = getCQInKV(topic, queueId, midCQOffset);
+            if (this.messageStore.getMessageStoreConfig().isEnableRocksDBLog()) {
+                ROCKSDB_LOG.warn("binarySearchInCQ. {}, {}, {}, {}, {}", topic, queueId, midCQOffset, low, high);
+            }
+            if (byteBuffer == null) {
+                low = midCQOffset + 1;
+                continue;
+            }
+
+            final long phyOffset = byteBuffer.getLong(PHY_OFFSET_OFFSET);
+            if (phyOffset == targetPhyOffset) {
+                if (min) {
+                    resultCQOffset =  midCQOffset;
+                    resultPhyOffset = phyOffset;
+                }
+                break;
+            } else if (phyOffset > targetPhyOffset) {
+                high = midCQOffset - 1;
+                if (min) {
+                    resultCQOffset = midCQOffset;
+                    resultPhyOffset = phyOffset;
+                }
+            } else {
+                low = midCQOffset + 1;
+                if (!min) {
+                    resultCQOffset = midCQOffset;
+                    resultPhyOffset = phyOffset;
+                }
+            }
+        }
+        return new PhyAndCQOffset(resultPhyOffset, resultCQOffset);
+    }
+
+    public static Pair<ByteBuffer, ByteBuffer> getCQByteBufferPair() {
+        ByteBuffer cqKey = ByteBuffer.allocateDirect(RocksDBConsumeQueueStore.MAX_KEY_LEN);
+        ByteBuffer cqValue = ByteBuffer.allocateDirect(CQ_UNIT_SIZE);
+        return new Pair<>(cqKey, cqValue);
+    }
+
+    private ByteBuffer buildCQKeyBB(final byte[] topicBytes, final int queueId, final long cqOffset) {
+        final ByteBuffer bb = ByteBuffer.allocate(CQ_KEY_LENGTH_WITHOUT_TOPIC_BYTES + topicBytes.length);
+        buildCQKeyBB0(bb, topicBytes, queueId, cqOffset);
+        return bb;
+    }
+
+    private void buildCQKeyBB(final ByteBuffer bb, final byte[] topicBytes, final int queueId, final long cqOffset) {
+        bb.position(0).limit(CQ_KEY_LENGTH_WITHOUT_TOPIC_BYTES + topicBytes.length);
+        buildCQKeyBB0(bb, topicBytes, queueId, cqOffset);
+    }
+
+    private void buildCQKeyBB0(final ByteBuffer bb, final byte[] topicBytes, final int queueId, final long cqOffset) {
+        bb.putInt(topicBytes.length).put(CTRL_1).put(topicBytes).put(CTRL_1).putInt(queueId).put(CTRL_1).putLong(cqOffset);
+        bb.flip();
+    }
+
+    private void buildCQValueBB(final ByteBuffer bb, final long phyOffset, final int msgSize, final long tagsCode, final long storeTimestamp) {
+        bb.position(0).limit(CQ_UNIT_SIZE);
+        buildCQValueBB0(bb, phyOffset, msgSize, tagsCode, storeTimestamp);
+    }
+
+    private void buildCQValueBB0(final ByteBuffer bb, final long phyOffset, final int msgSize,
```

Comment:
Suggest changing all 'BB' to "ByteBuffer"

Reply:
ok